[context] Use global ConfigMain::keepcache(), use value from config file

jrohel opened the pull request:
Previously, "keep_cache" was part of the DnfContextPrivate structure. And its default value was "false". The value in the configuration file was ignored.
After the fix, the value from the configuration file is used.
As a side effect, the current "keepcache" value is now shared between contexts.

jrohel committed:
[context] Use global ConfigMain::keepcache(), use value from config file

Previously, "keep_cache" was part of the DnfContextPrivate structure. And
its default value was "false". The value in the configuration file was ignored.
After the fix, the value from the configuration file is used.
As a side effect, the current "keepcache" value is now shared between contexts.

Conan-Kudo commented:
@rh-atomic-bot r+

lukash commented:
@rh-atomic-bot try

jrohel commented:
@rh-atomic-bot retry

lukash commented:
@rh-atomic-bot try

lukash commented:
@rh-atomic-bot try

jlebon commented:
@rh-atomic-bot ping

jlebon commented:
@rh-atomic-bot try

lukash commented:
bors try

Conan-Kudo commented:
Wait, the command is bors now?

lukash commented:
> Wait, the command is bors now?

Not entirely, it's not working yet, but yes, we are switching. I'll let you know when it's done.

jrohel commented:
I rebased the PR.

Conan-Kudo commented:
bors r+

lukash commented:
@Conan-Kudo libdnf failed to build, a unit test is failing: https://copr.fedorainfracloud.org/coprs/rpmsoftwaremanagement/rpm-gitoverlay-1611186776.701680/build/1890079/

Conan-Kudo commented:
@lukash Also, it occurs to me, is Bors set up correctly so it does rebase merges instead of merge commits?

lukash commented:
@Conan-Kudo it does merge commits :slightly_frowning_face: and can't be configured for rebases unless I'm missing something: https://github.com/bors-ng/bors-ng/issues/138

We are currently discussing this, your opinion? atomic bot is on its last legs, it's not really maintained anymore.

Conan-Kudo commented:
@lukash Maybe we should just use GitHub's own merging stuff again and just get Jenkins runs to be a required test on commit?

Conan-Kudo commented:
Another option might be @jd's @MergifyIO...

jd commented:
Let me know if you need to set it up. :)

lukash commented:
> Another option might be @jd's @Mergifyio...

Looking at it... but it's a paid service, right?

> @lukash Maybe we should just use GitHub's own merging stuff again and just get Jenkins runs to be a required test on commit?

Lets do that for now, while we decide what to do about this.

I somehow find myself reworking the whole CI setup on the go... how did that happen?

Conan-Kudo commented:
Mergify is free for OSS projects. Fedora already uses it for nearly all of its projects in fedora-infra GitHub organization.

lukash commented:
Okay, I'll check it out.

@jd thanks for the offer. Shooting you an email! :slightly_smiling_face: FWIW I could have clicked on the Pricing link but maybe you wanna mention it's free for OSS on the front page...

jrohel commented:
Unit tests have one big problem. They use the "dnf.conf" file from the host. This PR adds support for "keepcache" in the configuration file. If "keepcache=1" is in the host configuration, this will break the unit test.

I adjusted the failing test. However, the right solution for the future are unit tests that do not depend on the "dnf.conf"  on the host.

jrohel commented:
bors try

jrohel commented:
Unit tests passed. But CI does not. I'll check it.

jrohel committed:
[context] Don't read main configuration file in "dnf_context_set_*" methods

The "dnf_context_set_config_file_path" method had to be called before
the other "dnf_context_set_*" setters.
With this adjustment, it is no longer needed. The configuration file
is readed later - in "dnf_context_get_*".

jrohel commented:
try

jrohel commented:
bors try

jrohel commented:
The line `dnf_context_set_keep_cache (ctx, FALSE);` is hard-coded before calling the `dnf_context_set_config_file_path` method in microdnf. This was a problem, the configuration file was loaded from the default path before the path to the configuration file was set.
I improved the libdnf. Now the `dnf_context_set_ *` methods can be called before the `dnf_context_set_config_file_path` method is called.

jrohel commented:
Oh. I forgot one debug print.

jrohel commented:
bors try
